internal/npm_install/generate_build_file.ts

```diff
@@ -440,29 +462,35 @@ def _maybe(repo_rule, name, **kwargs):
     )
 `;
 
-  writeFileSync(`install_${workspace}.bzl`, bzlFile);
+  await writeFile(`install_${workspace}.bzl`, bzlFile);
 }
 
 /**
  * Generate build files for a scope.
  */
-function generateScopeBuildFiles(scope: string, pkgs: Dep[]) {
+async function generateScopeBuildFiles(scope: string, pkgs: Dep[]) {
   const buildFile = generateBuildFileHeader() + printScope(scope, pkgs);
-  writeFileSync(path.posix.join(scope, 'BUILD.bazel'), buildFile);
+  await writeFile(path.posix.join(scope, 'BUILD.bazel'), buildFile);
 }
 
 /**
  * Checks if a path is a file.
  */
-function isFile(p: string) {
-  return fs.existsSync(p) && fs.statSync(p).isFile();
+async function isFile(p: string) {
+  try {
+  return (await fs.stat(p)).isFile();
```

Comment:
Indentation.

Again this does the `catch` unlike previously?

Also why try/catch instead of `.catch` like the other spots?

Reply:
Same as isPkgDirASymlink.

Reply:
Lets make it the same as `isDirectory` though..? Doing `.catch(() => false)` instead of try/catch...

Reply:
OK